homeassistant/components/snmp/device_tracker.py

```diff
@@ -51,78 +71,119 @@ class SnmpScanner(DeviceScanner):
     """Queries any SNMP capable Access Point for connected devices."""
 
     def __init__(self, config):
-        """Initialize the scanner."""
-
-        self.snmp = cmdgen.CommandGenerator()
-
-        self.host = cmdgen.UdpTransportTarget((config[CONF_HOST], 161))
-        if CONF_AUTH_KEY not in config or CONF_PRIV_KEY not in config:
-            self.auth = cmdgen.CommunityData(config[CONF_COMMUNITY])
+        """Initialize the scanner and test the target device."""
+        host = config.get(CONF_HOST)
+        port = DEFAULT_PORT
```

Comment:
```suggestion
```

This is never mutated, please use the constant below instead

Reply:
As mentioned before, these temporary placeholders are there to standardize the config structure [across the SNMP integrations][snmp-switch-conf] and facilitate the transition to the UI config flow by just reverting 63e1c06 and continuing from there. I plan on tackling that task once the device_tracker has been fixed.

[snmp-switch-conf]: https://github.com/home-assistant/core/blob/57cc3495c2a26b7ba29d8f9f252238865e6132e1/homeassistant/components/snmp/sensor.py#L112

Reply:
We don't want to merge temporary code. The change should be made when the code is going to be used.

Reply:
All right, but then I'm going to leave the future work to someone else entirely. I'm not keen on doing and undoing my work three times :)

Reply:
We picked up Christian's work after 7 years in #113624, so maybe someone will pick yours in the future 👀

Reply:
Since you have indicated you don't want to followup on this, normally we going to need a codeowner to of this integration to accept responsibility for this change 
https://github.com/home-assistant/architecture/blob/master/adr/0008-code-owners.md#rules

Unfortunately there are no codeowners for this integration currently

Reply:
Let me rephrase that: I clearly take full ownership of these changes and intend to support them, I just don't plan to work on moving the SNMP integrations to the new UI config flow anymore.

My bad if that wasn't clear, I'm just puzzled as to why this PR is getting so much scrutiny when other changes related to the same broken component have been let through with ease. Especially since people have been waiting a month (and two releases) for this fix when it was ready all along.

Reply:
> My bad if my intents weren't clear,
>  I clearly take full ownership of these changes and intend to support them,

Excellent. please add yourself to `codeowners` in `manifiest.json`. Thanks

>  I'm just puzzled as to why this PR is getting so much scrutiny when other changes related to the same broken component have been let through with ease. Especially since people have been waiting a month (and two releases) for this fix when it was ready all along.

This PR started out with ADR0007 violations. Other PRs did not have this problem.